File: Assignment 3/Q14.c
#include <stdio.h>
int main()
{
    int i, j, fact1 = 1, fact2, fact3, k, space, num;
    for (i = 0; i <= 4; i++)
    {
        if (i == 0)
            fact1 = 1;
        else
            fact1 *= i;
        for (space = 1; space <= 4 - i; space++)
        {
            printf(" ");
        }
        fact2 = 1;
        for (j = 0; j <= i; j++)
        {
            if (j == 0)
                fact2 = 1;
            else
                fact2 *= j;
            fact3 = 1;
            if (i != j)
                for (k = i - j; k >= 1; k--)
                {
                    fact3 *= k;
                }
            else
                fact3 *= 1;
            num = fact1 / fact2 / fact3;
            printf(" %d", num);
        }
        printf("\n");
    }
}

File: Assignment 3/Q14.py

for i in range(5):
    fact1 = 1
    for x in range(1, i + 1):
        fact1 *= x
    for space in range(4 - i):
        print(" ", end="")
    for j in range(i + 1):
        fact2 = 1
        for y in range(1, j + 1):
            fact2 *= y
        fact3 = 1
        for z in range(1, i - j + 1):
            fact3 *= z
        num = fact1 // (fact2 * fact3)
        print(f" {num}", end="")
    print()
